fastlane_bot/events/exchanges/uniswap_v2.py

```diff
@@ -59,3 +58,16 @@ async def get_tkn0(address: str, contract: AsyncContract, event: Any) -> str:
     @staticmethod
     async def get_tkn1(address: str, contract: AsyncContract, event: Any) -> str:
         return await contract.caller.token1()
+
+    def get_pool_function(self, factory_contract: Contract):
+        """ Function to get pools from Factory.
+            This function is intended to be used with a Multicall. It fetches pools from a Uniswap V2 fork Factory contract.
+
+            Args:
+                factory_contract: The factory contract.
+
+            Returns:
+                The function.
+
+            """
+        return factory_contract.functions.getPair
```

Comment:
You may as well just use it directly where needed (since you're checking UniV2 before calling it).

Reply:
Not relevant any more